### PR TITLE
merge pomm bridge with pommbundle

### PR DESCRIPTION
**Pull request by stood**

### CODE REVIEW

**Comment by stood** [comment]
I add conflict but I have a pb with namespace I will to correct

**Comment by stood** [comment]
It's good :)

File: sources/lib/PropertyInfo/Extractor/TypeExtractor.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            $session = $this->pomm->getDefaultSession();
+        }
+
+        if (isset($context['model:name'])) {
+            $model_name = $context['model:name'];
+        } else {
+            $model_name = "${class}Model";
+        }
+
+        if (!class_exists($model_name)) {
```
Comment (Neirda24) [inline sources/lib/PropertyInfo/Extractor/TypeExtractor.php]: Might be worth to move it up a bit. As it doesn't require the session to exit.

File: sources/lib/PropertyInfo/Extractor/TypeExtractor.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                $type = Type::BUILTIN_TYPE_OBJECT;
+                $name = 'PommProject\Foundation\Converter\Type\NumRange';
+                break;
+            case 'TsRange':
+                $type = Type::BUILTIN_TYPE_OBJECT;
+                $name = 'PommProject\Foundation\Converter\Type\TsRange';
+                break;
+            default:
+                $type = Type::BUILTIN_TYPE_OBJECT;
+                $name = $pomm_type;
```
Comment (Neirda24) [inline sources/lib/PropertyInfo/Extractor/TypeExtractor.php]: `name` seems unused. `class` instead ?

File: sources/lib/Serializer/Normalizer/FlexibleEntityNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+class FlexibleEntityNormalizer implements NormalizerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($object, $format = null, array $context = array())
+    {
+        return $object->extract();
```
Comment (Neirda24) [inline sources/lib/Serializer/Normalizer/FlexibleEntityNormalizer.php]: I would build an `stdClass` and forward it to a the normalizer again. (By implementing `NormalizerAwareInterface`)

File: sources/lib/Configurator/DatabaseCollectorConfigurator.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    public function __construct(DataCollector $datacollector)
+    {
+        $this->datacollector = $datacollector;
+    }
+
+    /**
+     * @param Pomm $pomm
+     *
+     * @return null
```
Comment (Neirda24) [inline sources/lib/Configurator/DatabaseCollectorConfigurator.php]: return void

File: sources/lib/DatabaseDataCollector.php [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+ */
+class DatabaseDataCollector extends DataCollector
+{
+    /** @var Stopwatch */
+    private $stopwatch;
+
+    public function __construct($unused = null, Stopwatch $stopwatch = null)
+    {
+        if ($unused !== null) {
+            trigger_error("The parameter Pomm has been deleted for to delete the high dependency.", E_USER_DEPRECATED);
```
Comment (Neirda24) [inline sources/lib/DatabaseDataCollector.php]: Maybe rephrase it to `The Pomm parameter has been deleted in order to remove the higher dependency` or somethin ?

File: sources/lib/DatabaseDataCollector.php [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+            'exception' => null,
+        ];
+    }
+
+    /**
+     * @param string $name
+     * @param array $data
+     * @param $session
+     *
+     * @return null
```
Comment (Neirda24) [inline sources/lib/DatabaseDataCollector.php]: void

File: sources/lib/DatabaseDataCollector.php [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+            'time' => 0,
+            'queries' => [],
+            'exception' => null,
+        ];
+    }
+
+    /**
+     * @param string $name
+     * @param array $data
+     * @param $session
```
Comment (Neirda24) [inline sources/lib/DatabaseDataCollector.php]:
```php
@param Session $session
```

File: sources/lib/PropertyInfo/Extractor/ListExtractor.php [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        if (isset($context['session:name'])) {
+            $session = $this->pomm->getSession($context['session:name']);
+        } else {
+            $session = $this->pomm->getDefaultSession();
+        }
+
+        if (isset($context['model:name'])) {
+            $model_name = $context['model:name'];
+        } else {
+            $model_name = "${class}Model";
```
Comment (Neirda24) [inline sources/lib/PropertyInfo/Extractor/ListExtractor.php]: I prefer the 
```php
$model_name = "{$class}Model";
```
syntax as it allows the use of sub properties or function calls

File: sources/lib/PropertyInfo/Extractor/ListExtractor.php [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        }
+
+        if (!class_exists($model_name)) {
+            return;
```
Comment (Neirda24) [inline sources/lib/PropertyInfo/Extractor/ListExtractor.php]: Would add a warning in the logs could be usefull here ?

File: sources/lib/PropertyInfo/Extractor/TypeExtractor.php [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+        if (isset($context['session:name'])) {
+            $session = $this->pomm->getSession($context['session:name']);
+        } else {
+            $session = $this->pomm->getDefaultSession();
+        }
+
+        if (isset($context['model:name'])) {
+            $model_name = $context['model:name'];
+        } else {
+            $model_name = "${class}Model";
```
Comment (Neirda24) [inline sources/lib/PropertyInfo/Extractor/TypeExtractor.php]: same thing, I prefer the
```php
$model_name = "{$class}Model";
```
syntax as it allows the use of sub properties or function calls

File: sources/lib/PropertyInfo/Extractor/ListExtractor.php [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+            $session = $this->pomm->getDefaultSession();
+        }
+
+        if (isset($context['model:name'])) {
+            $model_name = $context['model:name'];
+        } else {
+            $model_name = "${class}Model";
+        }
+
+        if (!class_exists($model_name)) {
```
Comment (Neirda24) [inline sources/lib/PropertyInfo/Extractor/ListExtractor.php]: Might be worth to move it up a bit. As it doesn't require the session to exit.

File: sources/lib/Serializer/Normalizer/FlexibleEntityDenormalizer.php [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+        if (isset($context['session:name'])) {
+            $session = $this->pomm->getSession($context['session:name']);
+        } else {
+            $session = $this->pomm->getDefaultSession();
+        }
+
+        if (isset($context['model:name'])) {
+            $model_name = $context['model:name'];
+        } else {
+            $model_name = "${class}Model";
```
Comment (Neirda24) [inline sources/lib/Serializer/Normalizer/FlexibleEntityDenormalizer.php]: same thing 
```php
"{$class}Model";
```

File: sources/lib/Serializer/Normalizer/FlexibleEntityDenormalizer.php [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsDenormalization($data, $type, $format = null)
+    {
+        $reflection = new \ReflectionClass($type);
+        $interfaces = $reflection->getInterfaces();
+
+        // @TODO Use FlexibleEntityInterface::class with php >= 5.5
+        return isset($interfaces['PommProject\ModelManager\Model\FlexibleEntity\FlexibleEntityInterface']);
```
Comment (Neirda24) [inline sources/lib/Serializer/Normalizer/FlexibleEntityDenormalizer.php]: Couldn't we achieve the same result with [is_subclass_of](https://www.php.net/manual/fr/function.is-subclass-of.php) ?

**Comment by stood** [comment]
After reflexion, I'm wondering if it's a good solution. In the futur, if we want develop a new Bundle with a new component Pomm (other ModelManager for example) we will to duplicate the code.
